lib/autoproj/cli/standalone_ci.rb

```diff
@@ -0,0 +1,125 @@
+            end
+
+            desc 'dpkg-filter-status STATUS_PATH [RULES]',
+                 'outputs the list of APT packages to install based on a dpkg status '\
+                 'file and a set of inclusion/exclusion rules of the form "+ pattern" '\
+                 'and "- pattern"'
+            option :orig, desc: 'a status file whose installed packages are removed',
+                          type: :string
+            option :file, desc: 'read the rules from a file',
+                          type: :string
+            def dpkg_filter_status(status_path, *rules)
+                rules += File.readlines(options[:file]).map(&:strip) if options[:file]
+                rules = rules.map do |line|
+                    next if line.empty? || line.start_with?('#')
+
+                    parse_rule(line)
+                end
+
+                packages = Autoproj::CI::Rebuild.dpkg_create_package_install(
+                    status_path, rules, orig: options[:orig]
+                )
+                puts packages.join("\n")
+            end
+
+            no_commands do # rubocop:disable Metrics/BlockLength
+                def parse_rule(line)
+                    unless (m = /^([+-])\s+(.*)/.match(line))
+                        raise ArgumentError, "invalid rule line '#{line}'"
+                    end
+
+                    mode = (m[1] == '+')
+                    begin
+                        [mode, Regexp.new(m[2])]
+                    rescue RegexpError => e
+                        raise ArgumentError, "invalid regexp in '#{line}': #{e}"
+                    end
+                end
+
+                def prepare_workspace(config_dir, output_dir, workspace_dir)
+                    FileUtils.mkdir_p File.join(output_dir, workspace_dir, '.autoproj')
+
+                    if File.file?(envsh = File.join(config_dir, 'env.sh'))
+                        filter_envsh(envsh, output_dir, workspace_dir)
+                        generate_autoproj_stub(output_dir, workspace_dir)
+                    end
+
+                    if File.file?(file = File.join(config_dir, 'source.yml'))
+                        FileUtils.cp(
+                            file, File.join(output_dir, workspace_dir, '.autoproj')
+                        )
+                    end
+
+                    if File.file?(file = File.join(config_dir, 'installation-manifest'))
+                        FileUtils.cp(
+                            file,
+                            File.join(output_dir, workspace_dir, 'installation-manifest')
+                        )
+                    end
+                end
+
+                AUTOPROJ_STUB_PATH = File.join(__dir__, 'autoproj-stub.sh.erb').freeze
+
+                def filter_envsh(source_path, output_dir, workspace_dir)
+                    filtered = File.readlines(source_path)
+                                   .find_all { |l| !/^(source|\.)/.match?(l) }
```

Comment:
What if we added a `user: false` keyword argument to the `#source_after` (also `source_before` and possibly others) API to filter out only env files that were not explicitly added by the user?  Obviously, env files sourced with `user: true` would have to be copied to the build root tarball as well

Reply:
It might work ... but tbh I'd rather not spend the effort needed to make it work until I have the need. I, for one, do not ever have had to explicitly source files in my autoproj environments so ...

Reply:
> I, for one, do not ever have had to explicitly source files

I do, actually :( 

Nevermind if you think it's too much work (for the sake of argument, I don't think it is)